src/fparser/two/Fortran2003.py

```diff
@@ -5978,9 +5978,10 @@ class If_Stmt(StmtBase):  # R807
     """
     subclass_names = []
     use_names = ['Scalar_Logical_Expr', 'Action_Stmt_C802']
+    action_stmt_cls = Action_Stmt_C802
 
-    @staticmethod
-    def match(string):
+    @classmethod
```

Comment:
I see what you mean. I didn't take the fact that this was a staticmethod into account. However, I think your solution is good and removes a good chunk of replication. Sorry to be a pain (and in fact remove all vestiges of the original carrot) but please could you add a docstring for this method.

Reply:
Agreed, I think it serves also as a good blueprint for similar situations that will probably arise for other F2008 features when they are added. Will take care of the docstring.